pkg/gatherers/workloads/gather_helm_info.go

```diff
@@ -0,0 +1,129 @@
+
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// GatherHelmInfo Collects summarized info about the helm usage on a cluster
+// in a generic fashion
+//
+// ### API Reference
+// None
+//
+// ### Sample data
+// - docs/insights-archive-sample/config/helm_info.json
+//
+// ### Location in archive
+// - `config/helm_info.json`
+//
+// ### Config ID
+// `workloads/helm_info`
+//
+// ### Released version
+// - 4.15.0
+//
+// ### Backported versions
+// None
+func (g *Gatherer) GatherHelmInfo(ctx context.Context) ([]record.Record, []error) {
+	dynamicClient, err := dynamic.NewForConfig(g.gatherKubeConfig)
+	// dynamicClient, err := dynamic.NewForConfig(g.gatherProtoKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return gatherHelmInfo(ctx, dynamicClient)
+}
+
+func gatherHelmInfo(
+	ctx context.Context,
+	dynamicClient dynamic.Interface,
+) ([]record.Record, []error) {
+	// Define the Helm-related labels and annotations
+	helmLabels := map[string]string{
+		"app.kubernetes.io/managed-by": "Helm",
+	}
+	helmAnnotations := []string{
+		"helm.sh/chart",
+	}
+
+	// List ReplicaSets, DaemonSets, StatefulSets, Services, and Deployments in the namespace
+	resources := []schema.GroupVersionResource{
+		{Group: "apps", Version: "v1", Resource: "replicasets"},
+		{Group: "apps", Version: "v1", Resource: "daemonsets"},
+		{Group: "apps", Version: "v1", Resource: "statefulsets"},
+		{Group: "", Version: "v1", Resource: "services"},
+		{Group: "apps", Version: "v1", Resource: "deployments"},
+	}
+
+	var errs []error
+	var records []record.Record
+	for _, resource := range resources {
+
+		listOptions := metav1.ListOptions{}
+		//switch resource {
+		//case "services":
+		//	listOptions = metav1.ListOptions{
+		//		LabelSelector: fmt.Sprintf("%s=%s", helmLabels["app.kubernetes.io/managed-by"], "Helm"),
+		//	}
+		//}
+
+		items, err := dynamicClient.Resource(resource).List(ctx, listOptions)
+		if errors.IsNotFound(err) {
+			return nil, nil
+		}
+		if err != nil {
+			klog.V(2).Infof("Unable to list %s resource due to: %s", resource, err)
+			errs = append(errs, err)
+			continue
+		}
+
+		for _, item := range items.Items {
+			annotations := item.GetAnnotations()
+			if annotations == nil {
+				annotations = map[string]string{}
+			}
+
+			// Check if the resource has Helm-related annotations
+			helmInfo := map[string]string{}
+			for _, annotation := range helmAnnotations {
+				if value, ok := annotations[annotation]; ok {
+					helmInfo[annotation] = value
+				}
+			}
+
+			// Check if the resource has Helm-related labels
+			labels := item.GetLabels()
+			isHelmManaged := true
+			for key, value := range helmLabels {
+				if labels[key] != value {
+					isHelmManaged = false
+					break
+				}
+			}
+
+			if isHelmManaged {
+				records = append(records, record.Record{
+					Name: fmt.Sprintf("config/helm_info/%s/%s", resource.Resource, item.GetName()),
+					Item: record.JSONMarshaller{Object: item},
```

Comment:
Do we need to gather the full object definition? I guess this would be data explosion in the archive....